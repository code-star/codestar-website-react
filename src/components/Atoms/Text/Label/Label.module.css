.label {
  display: inline-block;
  padding: 0;
  text-align: left;
  font-family: 'metabold_roman', arial;
  background-color: #fff;
  user-select: none;
  background-color: #fff;
}
.labelGrey {
  color: #c6c6c6;
}
.labelBlack {
  color: #000;
}
.labelBlue {
  color: #00f;
}
.labelWhite {
  color: #fff;
}
.labelTiny {
  font-size: 8px;
}
.labelSmall {
  font-size: 14px;
}
.labelMedium {
  font-size: 16px;
}
.labelLarge {
  font-size: 30px;
}
