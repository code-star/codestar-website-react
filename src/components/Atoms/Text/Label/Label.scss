@import '../../../../theming/config.scss';

.label {
    display: inline-block;
    padding: 0;
    text-align: left;
    font-family: $font-family-primary;
    background-color: $color-white;
    user-select: none;

    &Grey {
        color: $color-grey;
    }

    &Black {
        color: $color-black;
    }

    &Blue {
        color: $color-blue;
    }

    &Tiny {
        font-size: $font-size-tiny;
    }

    &Small {
        font-size: $font-size-small;
    }

    &Medium {
        font-size: $font-size-medium;
    }

    &Large {
        font-size: $font-size-large;
    }

}
