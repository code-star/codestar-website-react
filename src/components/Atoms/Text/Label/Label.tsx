import React from 'react';
import classNames from 'classnames/bind';
import style from './Label.module.css';
import { EColor, ESize } from '../../../../utility/enumeration';

const cx = classNames.bind(style);

interface ILabelProps {
	color: EColor;
	size: ESize;
	text: string;
}

const Label = ({ color, size, text }: ILabelProps) => (
	<span
		className={cx(
			'label',
			{
				labelGrey: color === 'grey',
				labelBlack: color === 'black',
				labelBlue: color === 'blue',
			},
			{
				labelTiny: size === 'tiny',
				labelSmall: size === 'small',
				labelMedium: size === 'medium',
				labelLarge: size === 'large',
			}
		)}
	>
		{text}
	</span>
);

export default Label;
