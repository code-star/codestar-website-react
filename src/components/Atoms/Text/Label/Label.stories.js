import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, select } from '@storybook/addon-knobs';
import Label from './Label';

storiesOf('Components/Atoms/Text', module)
	.addDecorator(withKnobs)
	.add('Label', () => {
		const color = select('Color', ['grey', 'black', 'blue'], 'grey');

		const size = select('Size', ['tiny', 'small', 'medium', 'large'], 'grey');

		return <Label color={color} size={size} text="Awesome label" />;
	});
