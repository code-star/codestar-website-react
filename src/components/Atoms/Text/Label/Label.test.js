import React from 'react';
import { shallow } from 'enzyme';

import Label from './Label';

const getComponent = (color, size) => (
  <Label color={color} size={size} text="Awesome label" />
);

const renderShallow = () => {
  return shallow(getComponent('grey', 'medium'));
};

describe('<Label />', () => {
  let wrapper;

  describe('Instance', () => {
    test('must be an instance of Label', () => {
      wrapper = renderShallow();
      expect(wrapper.find('Label')).toBeTruthy();
    });
  });

  describe('Snaphot', () => {
    test('must match blue ~ tiny', () => {
      expect(
        global.renderToJSON(getComponent('blue', 'tiny'))
      ).toMatchSnapshot();
    });

    test('must match blue ~ small', () => {
      expect(
        global.renderToJSON(getComponent('blue', 'tiny'))
      ).toMatchSnapshot();
    });

    test('must match blue ~ medium', () => {
      expect(
        global.renderToJSON(getComponent('blue', 'medium'))
      ).toMatchSnapshot();
    });

    test('must match blue ~ large', () => {
      expect(
        global.renderToJSON(getComponent('blue', 'large'))
      ).toMatchSnapshot();
    });
  });
});
