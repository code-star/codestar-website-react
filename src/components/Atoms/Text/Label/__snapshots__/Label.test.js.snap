// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> Snaphot must match blue ~ large 1`] = `
<span
  className="label labelBlue labelLarge"
>
  Awesome label
</span>
`;

exports[`<Label /> Snaphot must match blue ~ medium 1`] = `
<span
  className="label labelBlue labelMedium"
>
  Awesome label
</span>
`;

exports[`<Label /> Snaphot must match blue ~ small 1`] = `
<span
  className="label labelBlue labelTiny"
>
  Awesome label
</span>
`;

exports[`<Label /> Snaphot must match blue ~ tiny 1`] = `
<span
  className="label labelBlue labelTiny"
>
  Awesome label
</span>
`;
